src/Database/ResultSet.php

```diff
@@ -140,8 +140,8 @@ public function normalizeRow(array $row): array
 				$row[$key] = is_float($tmp = $value * 1) ? $value : $tmp;
 
 			} elseif ($type === IStructure::FIELD_FLOAT) {
-				if (($pos = strpos($value, '.')) !== false) {
-					$value = rtrim(rtrim($pos === 0 ? "0$value" : $value, '0'), '.');
+				if (($pos = strpos((string) $value, '.')) !== false) {
```

Comment:
So what about `if (is_string($value) && ($pos = …)` ?

Reply:
That's even better, updated, thanks!